.github/workflows/publish.yml

```diff
@@ -1,19 +1,14 @@
 name: Publish to PyPI
 
-on:
-  push:
-    tags:
-      - "*"
+on: push
```

Comment:
Do I understand correctly that it will deploy to test PyPI on a push to any branch?

Reply:
Yes, that's what it means. We could limit to to 1.26.x and main, in order to avoid the dependabot pull requests to generate deploys.

Reply:
Then with the versioning scheme from #3412 it's possible that two releases can be created from different branches with the same number of commits after a release, right? If so, PyPI won't accept them

Reply:
I've pushed [`b6c02e1` (#3411)](https://github.com/urllib3/urllib3/pull/3411/commits/b6c02e11b3c6590e65fcffa42276b53a2ea9d91b) to only publish to Test PyPI for tags and the main/1.26.x branches. It remains to be seen if we'll adopt #3412 but doing that sounds like a good idea anyway. I also limited the GitHub Test PyPI Environment to main/1.26.x branches and tags.

Reply:
Nice, thanks!